### ISSUE TITLE
Repository is large

### ISSUE THREAD

**Issue by remixz** [hackclub/hackclub#744]
```bash
$ git clone https://github.com/hackclub/hackclub.git
Cloning into 'hackclub'...
remote: Counting objects: 4940, done.
remote: Compressing objects: 100% (71/71), done.
remote: Total 4940 (delta 36), reused 0 (delta 0), pack-reused 4869
Receiving objects: 100% (4940/4940), 245.78 MiB | 1.75 MiB/s, done.
Resolving deltas: 100% (2272/2272), done.
Checking connectivity... done.
```

Just want to track this for myself, and see if we can bring down this size in the future.

**Comment by zachlatta** [comment]
:+1:

/cc @thethirdone @paked

On Thu, Feb 25, 2016 at 11:04 AM, Zach Bruggeman <notifications@github.com>
wrote:

> $ git clone https://github.com/hackclub/hackclub.git
> Cloning into 'hackclub'...
> remote: Counting objects: 4940, done.
> remote: Compressing objects: 100% (71/71), done.
> remote: Total 4940 (delta 36), reused 0 (delta 0), pack-reused 4869
> Receiving objects: 100% (4940/4940), 245.78 MiB | 1.75 MiB/s, done.
> Resolving deltas: 100% (2272/2272), done.
> Checking connectivity... done.
>
> Just want to track this for myself, and see if we can bring down this size
> in the future.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/issues/744>.
>



-- 
Zach Latta
Hack Club | zach@hackclub.com

**Comment by TheThirdOne** [comment]
For reference #241, was about this before.

**Comment by TheThirdOne** [comment]
Ok, master is 135 mb zipped (190 mb unzipped) without any history.  Almost all of this is due to the workshops folder. @zachlatta, @remixz, is that a reasonable amout?

The workshops (sorted by size): portfolio, dodge, personal_website, thugify, soccer, and swiper are about 80% of the file size issues. Presumably due to gifs.

Whatever we do the history will still have the extra size unless we do rebasing ( for which I will not make a complicated rebasing command). Simply requiring all images to be externally hosted would fix the size issue here, and a simply rebase to remove all image files could be done.

**Comment by zachlatta** [comment]
How big of an issue is this realistically? I think there are a number of
advantages to having the images directly in the repo: chiefly relative link
checking from Markcop and the ability to do things entirely offline.

On Thu, Feb 25, 2016 at 6:57 PM, Benjamin Landers <notifications@github.com>
wrote:

> Ok, master is 135 mb zipped (190 mb unzipped) without any history. Almost
> all of this is due to the workshops folder. @zachlatta
> <https://github.com/zachlatta>, @remixz <https://github.com/remixz>, is
> that a reasonable amout?
>
> The workshops (sorted by size): portfolio, dodge, personal_website,
> thugify, soccer, and swiper are about 80% of the file size issues.
> Presumably due to gifs.
>
> Whatever we do the history will still have the extra size unless we do
> rebasing ( for which I will not make a complicated rebasing command).
> Simply requiring all images to be externally hosted would fix the size
> issue here, and a simply rebase to remove all image files could be done.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/issues/744#issuecomment-189087865>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

**Comment by TheThirdOne** [comment]
At least for me, I have to go make myself a cup of tea while I wait for the repo to clone. I think it is a significant barrier to entry for people that want to contribute. If we could cut the total clone size down to ~100 mb I think it would be fine

However, we would have to keep files sizes from creeping up. With #241, after the issue was closed, thugify, soccer and swiper are all major offenders. Markcop should have a size check built in.

Using shallow clones would also help.

**Comment by zachlatta** [comment]
What if we started by adding a command to shallow clone in a "Getting
Started" section in the README? Might make it easier to get people off on
the right foot.

On Thu, Feb 25, 2016 at 8:58 PM, Benjamin Landers <notifications@github.com>
wrote:

> At least for me, I have to go make myself a cup of tea while I wait for
> the repo to clone. I think it is a significant barrier to entry for people
> that want to contribute. If we could cut the total clone size down to ~100
> mb I think it would be fine
>
> However, we would have to keep files sizes from creeping up. With #241
> <https://github.com/hackclub/hackclub/issues/241>, after the issue was
> closed, thugify, soccer and swiper are all major offenders. Markcop should
> have a size check built in.
>
> Using shallow clones would also help.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/issues/744#issuecomment-189115484>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

**Comment by TheThirdOne** [comment]
That would at best get clone size down to 135 mb (probably more like 150 mb). I am also not sure about pushing shallow clones on beginners. They very easily could run into issues they won't understand how to fix.

The sizes of workshops really needs to be addressed. Shallow clones will only mask the problem for the short term. As the repo gets larger, the problem will continue, but become harder to solve.

The current files need to be shrunk / moved and future workshops need to not be merged if they are too big.

**Comment by zachlatta** [comment]
>
> The sizes of workshops really needs to be addressed. Shallow clones will
> only mask the problem for the short term. As the repo gets larger, the
> problem will continue, but become harder to solve.
>
Good point that shallow clones may cause issues beginners aren't sure how
to fix.

> The current files need to be shrunk / moved and future workshops need to
> not be merged if they are too big.
>
The problem will get worse over time, but I also think that
shrinking/moving the image assets (especially moving them) will be more
trouble than it's worth. Realistically, I don't think that waiting a few
extra minutes for the repo to clone will be a complete deal breaker for new
contributors. And the current benefit of not having to worry about an asset
host being online, if it's blocked by a school network, or whether the are
even correct in the first place is pretty big -- at least in my mind.

Maybe we could have a warning about the repo size in our CONTRIBUTING.md?

On Thu, Feb 25, 2016 at 9:10 PM, Benjamin Landers <notifications@github.com>
wrote:

> That would at best get clone size down to 135 mb (probably more like 150
> mb). I am also not sure about pushing shallow clones on beginners. They
> very easily could run into issues they won't understand how to fix.
>
> The sizes of workshops really needs to be addressed. Shallow clones will
> only mask the problem for the short term. As the repo gets larger, the
> problem will continue, but become harder to solve.
>
> The current files need to be shrunk / moved and future workshops need to
> not be merged if they are too big.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/issues/744#issuecomment-189117292>.
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>

**Comment by remixz** [comment]
After our discussion in the call (https://www.youtube.com/watch?v=IO-vVJlB9a0), here are our thoughts going forward:

* We do want to keep GIFs in the repository, since it's easier if the workshop needs to be held offline.
* We're going to put this discussion on hold for now, as our current focus is adding to our content, and encouraging community involvement. We don't want to add too many restrictions early on. We'll keep watching workshop sizes, and revisit this issue if it becomes a major problem.
* We want to add a section on cloning this repository with a shallow clone, so that people new to the repository can get started as quick as possible.

I'm going to close out this issue, as I think we've solved it for now. Please feel free to comment if you need anymore clarification.

**Comment by TheThirdOne** [comment]
For watching repo size in the future:
 ```
curl https://api.github.com/repos/hackclub/hackclub 2> /dev/null | grep size | tr -dc '[:digit:]'
```
gets the size (with history of the repo). 

The size of master can be obtained by downloading a zip.
